Lib/distutils/tests/test_spawn.py

```diff
@@ -124,6 +124,11 @@ def test_find_executable(self):
                     rv = find_executable(program)
                     self.assertEqual(rv, filename)
 
+    def test_spawn_missing_exe(self):
+        with self.assertRaises(DistutilsExecError) as ctx:
+            spawn(['does-not-exist'])
+            assert 'command does-no-exist failed' in str(ctx)
```

Comment:
I don't think that this assertion is ever executed. You need something like:

```suggestion
        with self.assertRaises(DistutilsExecError) as ctx:
            spawn(['does-not-exist'])
        self.assertIn('command does-no-exist failed', str(ctx.exception))
```

Reply:
Good catch. I remember being surprised when the test passed the first time. Fixed in 1b7cf1abce.

Reply:
"Good catch. I remember being surprised when the test passed the first time."

:-)